src/main/java/liquibase/ext/otbo/changes/CreateFlexibleViewGenerator.java

```diff
@@ -76,4 +83,33 @@ public String apply( Object o ) {
 
 		return sequel.toArray( new Sql[0] );
 	}
+	
+	private String addSchemaPrefixToFunctionNames( String sql, Database database ) {
```

Comment:
In MSSQL server you can have multiple schemas within the same database, it's really another way of partitioning data. 

i.e. 
* _[database_name].[schema_name].function_name("value")_

It's therefore possible to have two functions with the same name and signature in the same database.

e.g. 
* _[database_name].[schema_a].function_name("value")_
* _[database_name].[schema_b].function_name("value")_

I think there needs to be some kind of priority given to the schema that matches the views existing schema.

i.e. _[database_name].[schema_a].[view_name]_ uses _[database_name].[schema_a].function_name("value")_ instead of _[database_name].[schema_b].function_name("value")_ as the latter has a matching schema, so therefore it's the most exact match. 

In the case where there are multiple matches and none are in the same schema, then I suppose we should fall back to functions in the same database first, then dbo, then in-built functions. We need some kind of priority order.

Usually customers put the tables, views, functions etc into an opentwins schema. However, internally to make things difficult for ourselves we put the tables, views, etc. into the dbo schema and functions for consistency go into the opentwins schema. We therefore need it to be clever enough to cater for these inconsistencies.

Does that make sense?

Reply:
That makes sense. So, if there are multiple functions with the same name, priority should be given to the one in the same schema as the view, followed by the one in the dbo schema (if there is one). Should the opentwins schema be considered in the priority order? What if none of them are in any of the prioritised schema? The sys.objects table contains only user-defined objects, so references to in-built functions won't be affected.

Reply:
At customers, the database tables and functions would all be in the same schema, so it would naturally have precedence.

I think the issue only really arises for the development databases where the default schema is dbo, which is where the tables are stored, but we create the functions explicitly into an opentwins schema. 

I therefore think priority should be this:

1. opentwins schema
2. current schema
3. other schemas (priority if duplicates are hard to suggest, take the top result I suppose?)

Reply:
That sounds sensible. I'll take that approach and use the first result if there are still several with the same name and signature.